main.js

```diff
@@ -0,0 +1,186 @@
+const getquizNum = document.getElementById('quizNum');
+const getcategory = document.getElementById('category');
+const getdifficulty = document.getElementById('difficulty');
+const getquestion = document.getElementById('question');
+const getanswers = document.getElementById('answers').children;
+
+// スタート画面表示の関数
+const getStart = () => {
+  getquizNum.innerHTML = 'ようこそ';
+  getcategory.innerHTML = '';
+  getdifficulty.innerText = '';
+  getquestion.innerHTML = '以下のボタンをクリック';
+  getanswers[0].innerHTML = '開始';
+  getanswers[1].style.display = 'none';
+  getanswers[2].style.display = 'none';
+  getanswers[3].style.display = 'none';
+
+  document.getElementById('startClick').addEventListener('click', (e) => {
+    document.getElementById('startClick').removeAttribute('id');
+    getanswers[0].style.display = 'none';
+    waitApi();
+    getQuizArray();
+  });
+};
+
+
+
+// スタート後、非同期処理中の表示の関数
+const waitApi = () => {
+  getquizNum.innerHTML = '取得中';
+  getanswers[1].style.display = 'none';
+  getcategory.innerHTML = '';
+  getdifficulty.innerText = '';
+  getquestion.innerHTML = '少々お待ちください';
+};
+
+// クイズ終了後、スタート画面表示の関数
+const resetQuiz = () => {
+  getquizNum.innerHTML = 'ようこそ';
+  getcategory.innerHTML = '';
+  getdifficulty.innerText = '';
+  getquestion.innerHTML = '以下のボタンをクリック';
+  getanswers[1].innerHTML = '開始';
+  getanswers[1].setAttribute('id', 'resetClick');
+  getanswers[1].style.display = 'block';
+  getanswers[0].style.display = 'none';
+  getanswers[2].style.display = 'none';
+  getanswers[3].style.display = 'none';
+
+  document.getElementById('resetClick').addEventListener('click', (e) => {
+    document.getElementById('resetClick').removeAttribute('id');
+    getanswers[0].style.display = 'none';
+    console.log('idをとります');
+    waitApi();
+    getQuizArray();
+  });
+};
+
+// クイズデータ取得
+const getQuizArray = async () => {
+  const url = 'https://opentdb.com/api.php?amount=10';
+  const data = await fetch(url)
+    .then((res) => {
+      console.log('成功しました');
+      return res.json();
+    })
+    .catch((error) => {
+      console.log('失敗しました');
+    });
```

Comment:
せっかく `async` を使っているので、以下のようにして
他言語と似た見た目にするとモダンでわかりやすくなるかと。

```suggestion
  const url = 'https://opentdb.com/api.php?amount=10';
  let data;
  try {
    data = await (await fetch(url)).json();
  } catch (error) {
    console.log(`失敗しました => ${error}`);
  }
```